File: src/Modules/MasterData/MasterData.Contracts/Seeding/WellKnownMasterDataIds.cs
namespace MasterData.Contracts.Seeding;

/// <summary>Stable identifiers for system-seeded MasterData rows that other modules may reference.</summary>
public static class WellKnownMasterDataIds
{
    public static readonly Guid AdHocOperationType = new("30000000-0000-0000-0000-000000000001");

    public static readonly Guid AircraftPerLandingService = new("40000000-0000-0000-0000-000000000001");

    public static readonly Guid UnknownCustomer = new("50000000-0000-0000-0000-000000000001");

    public static readonly Guid UnknownService = new("40000000-0000-0000-0000-000000000003");
    public static readonly Guid UnknownTool = new("60000000-0000-0000-0000-000000000001");
    public static readonly Guid UnknownMaterial = new("70000000-0000-0000-0000-000000000001");
    public static readonly Guid UnknownGeneralSupport = new("80000000-0000-0000-0000-000000000001");
}

File: src/Host/OperationsSystem.Mobile/app/src/main/java/com/nags/operations/data/WellKnownMasterDataIds.kt
package com.nags.operations.data

/**
 * Stable MasterData identifiers mirrored from the server contract.
 *
 * Keep these values aligned with `MasterData.Contracts.Seeding.WellKnownMasterDataIds`.
 */
internal object WellKnownMasterDataIds {
    const val UnknownService = "40000000-0000-0000-0000-000000000003"
    const val UnknownTool = "60000000-0000-0000-0000-000000000001"
    const val UnknownMaterial = "70000000-0000-0000-0000-000000000001"
    const val UnknownGeneralSupport = "80000000-0000-0000-0000-000000000001"

    fun isUnknownResource(id: String): Boolean =
        listOf(UnknownTool, UnknownMaterial, UnknownGeneralSupport).any { it.equals(id, true) }

    const val UnknownCustomer = "50000000-0000-0000-0000-000000000001"
}
